compiler/AST/iterator.cpp

```diff
@@ -846,27 +896,47 @@ static void insertLocalsForRefs(Vec<Symbol*>& syms, FnSymbol* fn,
       CallExpr* move = toCallExpr(defs->v[0]->parentExpr);
       INT_ASSERT(move->isPrimitive(PRIM_MOVE));
 
-      SymExpr* se = toSymExpr(move->get(2));
-      CallExpr* call = toCallExpr(move->get(2));
-      if (se) {
+      if (SymExpr* se = toSymExpr(move->get(2)))
+      {
+        // The symbol is defined through a bitwise (pointer) copy.
         INT_ASSERT(se->var->type->symbol->hasFlag(FLAG_REF));
         if (se->var->defPoint->parentSymbol == fn) {
           syms.add_exclusive(se->var);
         }
-      } else if (call->isPrimitive(PRIM_ADDR_OF) ||
-                 call->isPrimitive(PRIM_GET_MEMBER) ||
-                 call->isPrimitive(PRIM_GET_MEMBER_VALUE)) {
-        SymExpr* rhs = toSymExpr(call->get(1));
-        syms.add_exclusive(rhs->var);
-      } else if (FnSymbol* fn = call->isResolved()) {
-        for_actuals(actual, call) {
-          SymExpr* se = toSymExpr(actual);
-          if (se->var->defPoint->parentSymbol == fn) {
-            syms.add_exclusive(se->var);
+      }
+      else if (CallExpr* call = toCallExpr(move->get(2)))
+      {
+        // The RHS is a call.
```

Comment:
Might explicitly say "Ths RHS is a function call" since 'call' is sort of ambiguous